File: profile_manager.py
import requests

BASE_URL = "http://localhost/backend/index.php"

def create_profile(name, email, age):
    response = requests.post(f"{BASE_URL}?action=create", json={
        "name": name,
        "email": email,
        "age": age
    })
    if response.status_code == 200:
        print("Profile created successfully.")
    else:
        print("Error:", response.json())

def clone_profile(profile_id):
    response = requests.post(f"{BASE_URL}?action=clone", json={"id": profile_id})
    if response.status_code == 200:
        print("Profile cloned successfully.")
    else:
        print("Error:", response.json())

# Example usage:
if __name__ == "__main__":
    print("Creating a new profile...")
    create_profile("John Doe", "john@example.com", 30)
    
    print("Cloning the profile with ID 1...")
    clone_profile(1)

File: frontend/src/components/ProfileManager.js
import React, { useState } from 'react';
import axios from 'axios';

function ProfileManager() {
    const [profile, setProfile] = useState({ name: '', email: '', age: '' });
    const [profileId, setProfileId] = useState('');

    const createProfile = async () => {
        try {
            await axios.post('http://localhost/prototype-pattern/index.php?action=create', profile, {
                headers: {
                    'Content-Type': 'application/json',
                }
            });
            alert("Profile created!");
        } catch (error) {
            console.error("Error creating profile:", error);
            alert("Failed to create profile");
        }
    };

    const cloneProfile = async () => {
        try {
            await axios.post('http://localhost/prototype-pattern/index.php?action=clone', { id: profileId }, {
                headers: {
                    'Content-Type': 'application/json',
                }
            });
            alert("Profile cloned!");
        } catch (error) {
            console.error("Error cloning profile:", error);
            alert("Failed to clone profile");
        }
    };

    return (
        <div>
            <h2>Create Profile</h2>
            <input placeholder="Name" onChange={(e) => setProfile({...profile, name: e.target.value})} />
            <input placeholder="Email" onChange={(e) => setProfile({...profile, email: e.target.value})} />
            <input placeholder="Age" type="number" onChange={(e) => setProfile({...profile, age: e.target.value})} />
            <button onClick={createProfile}>Create Profile</button>

            <h2>Clone Profile</h2>
            <input placeholder="Profile ID" onChange={(e) => setProfileId(e.target.value)} />
            <button onClick={cloneProfile}>Clone Profile</button>
        </div>
    );
}

export default ProfileManager;
